purchase_stock_picking_invoice_link/models/account_invoice.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2019 Vicent Cubells <vicent.cubells@tecnativa.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    @api.multi
+    def _prepare_invoice_line_from_po_line(self, line):
+        vals = super()._prepare_invoice_line_from_po_line(line)
+        moves = self.env['stock.move'].search([
+            ('purchase_line_id', '=', line.id),
+        ])
+        if moves:
+            move_ids = moves._get_moves()
+            vals['move_line_ids'] = [(6, 0, move_ids.ids)]
+            pickings = move_ids.mapped('picking_id')
+            pickings.invoice_ids = [(4, self.id)]
```

Comment:
Here, we should do the same as below, loop in pickings for writing invoice_ids, it fails with partial pickings for the same line.